core/src/main/java/com/radixdlt/consensus/vertexstore/VertexStoreJavaImpl.java

```diff
@@ -95,47 +97,64 @@ public final class VertexStoreJavaImpl implements VertexStore {
 
   private final Hasher hasher;
   private final Ledger ledger;
+  private final Serialization serialization;
+  private final Metrics metrics;
+  private final VertexStoreConfig config;
 
   private final Map<HashCode, VertexWithHash> vertices = new HashMap<>();
-  private final Map<HashCode, Set<HashCode>> vertexChildren = new HashMap<>();
+  private final Multimap<HashCode, HashCode> vertexChildren = HashMultimap.create();
   private final Map<HashCode, ExecutedVertex> executedVertices = new HashMap<>();
 
   // These should never be null
   private VertexWithHash rootVertex;
   private HighQC highQC;
-
-  private VertexStoreJavaImpl(
-      Ledger ledger, Hasher hasher, VertexWithHash rootVertex, HighQC highQC) {
+  private int currentSerializedSizeBytes;
+
+  public VertexStoreJavaImpl(
```

Comment:
This is all much nicer / easier to follow now. The consolidation of logic in `resetToState` and `recordHighQcOrVertexChange`; and use of explicit error codes makes this much easier to understand and reason about.